src/main/java/com/commercetools/helper/mapper/impl/payment/ProductNameMapper.java

```diff
@@ -37,21 +35,27 @@ public void setPrefixProductNameWithAttr(String prefixProductNameWithAttr) {
      */
     public String getPaypalItemName(@Nonnull LineItem lineItem, @Nonnull List<Locale> locales) {
         String productName = lineItem.getName().get(locales);
+        if (StringUtils.isBlank(productName)) {
+            Locale fallBackLocale = lineItem.getName().getLocales()
+                    .iterator().next();
+            productName = lineItem.getName().get(fallBackLocale);
+        }
         if (StringUtils.isNotBlank(prefixProductNameWithAttr)) {
-            String withPrefix = getPrefixForProductName(lineItem, locales) + productName;
-            return withPrefix.length() < PRODUCT_NAME_MAX_LENGTH ? withPrefix : productName;
+            String prefix = getPrefixForProductName(lineItem, locales);
+            if (StringUtils.isNotBlank(prefix)) {
+                return prefix + StringUtils.SPACE + productName;
```

Comment:
Why not overriding here productName instead of having multiple `return` statements?

Reply:
We have 2 return method. Either to append the prefix or not. I see this current method is a pure function and mutations on the original object would looks weird. Because no other method has mutated the ctp lineItem while constructing the paypal Item object.